src/applications/upgradeable_contract/upgradeable_contract.md

```diff
@@ -0,0 +1,14 @@
+# Upgradeable Contract
+
+Starknet distinguishes between a contract and its implementation by separating contracts into classes and instances.
+
+A contract class is the definition of the contract: Cairo byte code, hint information, entry point names, and everything that defines its semantics unambiguously.
+
+Each class is identified by its class hash, which is analogous to a class name in an object-oriented programming language. A contract instance is a deployed contract corresponding to a class.
+
+You can upgrade the class hash of a deployed contract by calling the `replace_class_syscall`. This example shows two contracts and how you can make them upgradeable. Try to deploy `UpgradeableContract_V0`, and then upgrade it to `UpgradeableContract_V1`.
```

Comment:
I would add another sentence after the first one to emphasize on the fact that this is what allow to upgrade the implementation of a contract:
"You can upgrade a deployed contract to a newer version by calling the `replace_class_syscall` function. By using this function, you can update the class hash associated with a deployed contract, effectively upgrading its implementation."

Reply:
And for the last part detailing the example, maybe start as a new paragraph to separate the theory and the example explanation. I would go further in the explanation to ensure better comprehension, like this:
"To illustrate this concept, let's consider an example with two contracts: `UpgradeableContract_V0` and `UpgradeableContract_V1`. Begin by deploying `UpgradeableContract_V0` as the initial version. Next, call `replace_class_syscall` to upgrade the class hash of the deployed contract to the one refering to the `UpgradeableContract_V1` implementation. Then, call the `version` function to see that the contract was upgraded to the V1 version."